drivers/spi/spi_ll_stm32.c

```diff
@@ -77,6 +106,19 @@ static void dma_callback(struct device *dev, void *arg,
 		} else if (channel == data->dma_rx.channel) {
 			/* this part of the transfer ends */
 			data->status_flags |= SPI_STM32_DMA_RX_DONE_FLAG;
+#if CONFIG_SPI_STM32_UNALIGNED_DMA
```

Comment:
Why is the cache invalidation done only when `CONFIG_SPI_STM32_UNALIGNED_DMA` is enabled? The cache also needs to be invalidated in case the buffers are assumed to be aligned.

Reply:
The Idea was that if the user does not enable the CONFIG_SPI_STM32_UNALIGNED_DMA option he is responsible for cache management, or by invalidating his buffers after SPI is done or by using uncachable memory. Because the driver can not know if an aligned buffer of for example 3 bytes can be invalidated, it could only invalidate buffers that are exactly correct aligned and have a multiple of the cache line size, which will give most certainly weird bug reports, like; "when I send 32 bytes it works, but when I send 48 half my data is not there"

Reply:
This is not clear from the description of the option, which seems to imply that things should just work if the buffer are aligned. 
Also I am not fully sure, but I think that you can actually need to use an extra buffer for TX, as long as all the cache lines containing the data are flushed. The extra cache flushes should just make a small performances impact. For RX and cache invalidation, the buffer must definitely be aligned on the cache line addresses and size to avoid data losses (or just ensure that no other data is on the affected cache lines, which is basically the same).

Reply:
The idea was that a cache flushes could over write memory that was written by another RX DMA, but when RX DMA is always aligned that can never happen. Even if they use the same buffer the flush would happen before the RX DMA starts. So the TX buffer could be removed.

On the other hand when one would also want to support using non-DMA-able memory the write buffer must be copied to memory that is DMA-able first.

Like you mentioned the Kconfig name is confusing and the whole thing will need way more discussion, I really think it is better to move it to its own PR.